### PR TITLE
test(sbb-toggle, sbb-toggle-check): add visual spec

### PR DESCRIPTION
**Pull request by MarioCastigliano**

### CODE REVIEW

File: src/elements/toggle-check/toggle-check.visual.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        it(
+          state.name,
+          state.with(async (setup) => {
+            await setup.withFixture(html` <sbb-toggle-check checked> Label </sbb-toggle-check> `);
+          }),
+        );
+      }
+    });
+
+    describe('without label', () => {
```
Comment (jeripeierSBB) [inline src/elements/toggle-check/toggle-check.visual.spec.ts:60]: For test cases with only one 'it' you dont need a describe. You can directly set it as title of 'it' also state (default here) is not needed

File: src/elements/toggle-check/toggle-check.visual.spec.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            <sbb-toggle-check checked
+              >Label
+              <sbb-icon slot="icon" name="eye-small"></sbb-icon>
+            </sbb-toggle-check>
+          `);
+        }),
+      );
+    });
+
+    describeEach({ checked: [false, true] }, ({ checked }) => {
```
Comment (jeripeierSBB) [inline src/elements/toggle-check/toggle-check.visual.spec.ts:105]: for single arrays you can just us a for of loop

File: src/elements/toggle-check/toggle-check.visual.spec.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        }),
+      );
+    });
+
+    describe('block variant', () => {
+      it(
+        visualDiffDefault.name,
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(html`
+            <div>
```
Comment (jeripeierSBB) [inline src/elements/toggle-check/toggle-check.visual.spec.ts:127]: The outer div should not be needed as it lands anyways in a div.
```suggestion
```

File: src/elements/toggle/toggle-option/toggle-option.visual.spec.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  visualDiffDefault,
+  visualRegressionFixture,
+} from '../../core/testing/private.js';
+
+import './toggle-option.js';
+
+describe(`sbb-toggle-option`, () => {
+  let root: HTMLElement;
+
+  const cases = {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle-option/toggle-option.visual.spec.ts:15]: For simple arrays you can directly use an array without a complex object. And then using a for of loop

File: src/elements/toggle/toggle-option/toggle-option.visual.spec.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { html, nothing } from 'lit';
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle-option/toggle-option.visual.spec.ts:1]: Wondering if it is worth to have separate toggle option tests at all, or if the toggle tests are sufficient? What do you think?

File: src/elements/toggle/toggle-option/toggle-option.visual.spec.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+describe(`sbb-toggle-option`, () => {
+  let root: HTMLElement;
+
+  const cases = {
+    label: [true, false],
+  };
+
+  describeViewports({ viewports: ['zero'] }, () => {
+    describe('default', () => {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle-option/toggle-option.visual.spec.ts:20]: You can directly use 'it' without 'describe' (see other comment).

File: src/elements/toggle/toggle-option/toggle-option.visual.spec.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+          <sbb-toggle-option value="Value 1">
+            ${label ? `Option` : nothing}
+            <sbb-icon name="app-icon-small" slot="icon"></sbb-icon>
+          </sbb-toggle-option>
+        `);
+      });
+
+      it(
+        `slotted icon ${visualDiffDefault.name}`,
+        visualDiffDefault.with((setup) => {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle-option/toggle-option.visual.spec.ts:60]: I think here I'd directly use await setup.withFixture and remove the beforeEach block (as it has only this test)

File: src/elements/toggle/toggle/toggle.visual.spec.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        it(
+          state.name,
+          state.with((setup) => {
+            setup.withSnapshotElement(root);
+          }),
+        );
+      }
+    });
+
+    describeEach(sizeCases, ({ size }) => {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle/toggle.visual.spec.ts:55]: Could be simplified to a array with a for of loop and without beforeEach block.

File: src/elements/toggle/toggle/toggle.visual.spec.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+      it(
+        `disabled ${visualDiffDefault.name}`,
+        visualDiffDefault.with((setup) => {
+          setup.withSnapshotElement(root);
+        }),
+      );
+    });
+
+    describeEach(iconCases, ({ size, label }) => {
+      beforeEach(async function () {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle/toggle.visual.spec.ts:71]: beforeEach could be removed and used with setup.withFixture

File: src/elements/toggle/toggle/toggle.visual.spec.ts [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+      it(
+        `icon ${visualDiffDefault.name}`,
+        visualDiffDefault.with((setup) => {
+          setup.withSnapshotElement(root);
+        }),
+      );
+    });
+
+    describe('slotted icon', () => {
```
Comment (jeripeierSBB) [inline src/elements/toggle/toggle/toggle.visual.spec.ts:85]: describe block not needed.

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Nice work! I have some little suggestions and some questions:
- Could you try to mimic this test as well: https://lyne-storybook.app.sbb.ch/?path=/story/elements-sbb-toggle-sbb-toggle--dynamic-width ?
- Could you add focus tests for the block variant and long label tests?

File: src/elements/toggle/toggle-option/toggle-option.visual.spec.ts (hunk 5 above)
Comment (MarioCastigliano) [reply src/elements/toggle/toggle-option/toggle-option.visual.spec.ts:1]: Since there are only a few cases, I don't think there's any harm in keeping them

**Review by jeripeierSBB — APPROVED** [review]
Very nice :)

**Pull request merged**